crates/re_renderer/src/line_drawable_builder.rs

```diff
@@ -14,44 +15,46 @@ use crate::{
 /// Builder for a vector of line strips, making it easy to create [`crate::renderer::LineDrawData`].
 ///
 /// TODO(andreas): We could make significant optimizations here by making this builder capable
-/// of writing to a GPU readable memory location.
-/// This will require some ahead of time size limit, but should be feasible.
-/// But before that we first need to sort out cpu->gpu transfers better by providing staging buffers.
-pub struct LineStripSeriesBuilder {
-    pub vertices: Vec<LineVertex>,
-
-    pub batches: Vec<LineBatchInfo>,
-
-    pub strips: Vec<LineStripInfo>,
+/// of writing to a GPU readable memory location for all its data.
+pub struct LineDrawableBuilder {
+    pub(crate) vertices: Vec<LineVertex>,
+    pub(crate) batches: Vec<LineBatchInfo>,
+    pub(crate) strips: Vec<LineStripInfo>,
 
     /// Buffer for picking instance id - every strip gets its own instance id.
     /// Therefore, there need to be always as many picking instance ids as there are strips.
     pub(crate) picking_instance_ids_buffer: CpuWriteGpuReadBuffer<PickingLayerInstanceId>,
 
     pub(crate) radius_boost_in_ui_points_for_outlines: f32,
-}
 
-impl LineStripSeriesBuilder {
-    pub fn new(ctx: &RenderContext) -> Self {
-        const RESERVE_SIZE: usize = 512;
+    max_num_strips: usize,
+    max_num_vertices: usize,
+}
 
-        // TODO(andreas): Be more resourceful about the size allocated here. Typically we know in advance!
+impl LineDrawableBuilder {
+    pub fn new(ctx: &RenderContext, max_num_strips: u32, max_num_vertices: u32) -> Self {
```

Comment:
This needs a docstring, i.e. why shouldn't I set max to `u32::MAX`?

Reading the code it looks like `max_num_strips` is used to pre-allocate VRAM, but `max_num_vertices` isn't (yet?)